cmd/kube-egress-cni/main.go

```diff
@@ -190,67 +189,74 @@ func cmdAdd(args *skel.CmdArgs) error {
 	if err != nil {
-		return err
+		logger.Error(err, "failed to parse CNI config")
+		return nil
 	}
 
 	// get k8s metadata
 	k8sInfo, err := conf.LoadK8sInfo(args.Args)
 	if err != nil {
-		return err
-	}
-	conn, err := grpc.NewClient(config.SocketPath,
-		grpc.WithTransportCredentials(insecure.NewCredentials()),
-		grpc.WithStreamInterceptor(grpc_retry.StreamClientInterceptor()),
-		grpc.WithUnaryInterceptor(grpc_retry.UnaryClientInterceptor()),
-		grpc.WithUnaryInterceptor(grpc_prometheus.UnaryClientInterceptor),
-		grpc.WithStreamInterceptor(grpc_prometheus.StreamClientInterceptor),
-		grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
-			d := net.Dialer{}
-			return d.DialContext(ctx, "tcp", addr)
-		}))
-	if err != nil {
-		return err
+		logger.Error(err, "failed to load k8s metadata")
+		return nil
 	}
-	defer conn.Close()
-	client := v1.NewNicServiceClient(conn)
-
 	podNs, err := ns.GetNS(args.Netns)
 	if err != nil {
-		return err
+		logger.Error(err, "failed to get pod namespace")
+		return nil
 	}
 	defer podNs.Close()
 	err = podNs.Do(func(nn ns.NetNS) error {
-
-		_, err = client.NicDel(context.Background(), &v1.NicDelRequest{
-			PodConfig: &v1.PodInfo{
-				PodName:      string(k8sInfo.K8S_POD_NAME),
-				PodNamespace: string(k8sInfo.K8S_POD_NAMESPACE),
-			},
-		})
-		if err != nil {
-			return err
-		}
-
 		ifHandle, err := netlink.LinkByName(consts.WireguardLinkName)
 		if err != nil {
 			//ignore error because cni delete may be invoked more than once.
 			return nil
 		}
-		return netlink.LinkDel(ifHandle)
+		if err := netlink.LinkDel(ifHandle); err != nil {
+			logger.Error(err, "failed to delete wireguard link")
+		}
+		return nil
 	})
 	if err != nil {
-		return err
+		logger.Error(err, "failed to delete wireguard link")
+		return nil
 	}
 
 	err = ipam.New(config.IPAM.Type, args.StdinData).DeleteIP()
 	if err != nil {
-		return err
+		logger.Error(err, "failed to delete ip")
+		return nil
+	}
+	conn, err := grpc.NewClient(config.SocketPath,
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithStreamInterceptor(grpc_retry.StreamClientInterceptor()),
+		grpc.WithUnaryInterceptor(grpc_retry.UnaryClientInterceptor()),
+		grpc.WithUnaryInterceptor(grpc_prometheus.UnaryClientInterceptor),
+		grpc.WithStreamInterceptor(grpc_prometheus.StreamClientInterceptor),
+		grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
+			d := net.Dialer{}
+			return d.DialContext(ctx, "tcp", addr)
+		}))
+	if err != nil {
+		logger.Error(err, "failed to contact cni manager daemon")
+		return nil
+	}
+	defer conn.Close()
+	client := v1.NewNicServiceClient(conn)
+	_, err = client.NicDel(context.Background(), &v1.NicDelRequest{
+		PodConfig: &v1.PodInfo{
+			PodName:      string(k8sInfo.K8S_POD_NAME),
+			PodNamespace: string(k8sInfo.K8S_POD_NAMESPACE),
+		},
+	})
+	if err != nil {
+		logger.Error(err, "failed to send nicDel request")
```

Comment:
how do we cleanup them later since error is ignored?

Reply:
it should be done in cni manager

Reply:
oh, the daemonset has done the cleanup job.

Reply:
That's great!